fedimint-core/src/encoding/mod.rs

```diff
@@ -611,6 +614,139 @@ impl Decodable for Cow<'static, str> {
     }
 }
 
+/// A type that decodes `module_instance_id`-prefixed `T`s even
+/// when corresponding `Decoder` is not available.
+///
+/// All dyn-module types are encoded as:
+///
+/// ```norust
+/// module_instance_id | len_u64 | data
+/// ```
+///
+/// So clients that don't have a corresponding module, can read
+/// the `len_u64` and skip the amount of data specified in it.
+///
+/// This type makes it more convenient. It's possible to attempt
+/// to retry decoding after more modules become available by using
+/// [`DynRawFallback::redecode_raw`].
+///
+/// Notably this struct does not ignore any errors. It only skips
+/// decoding when the module decoder is not available.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub enum DynRawFallback<T> {
```

Comment:
Building the fallback struct as a wrapper is a way more elegant solution than I had in mind!